src/SimulationRunner.cc

```diff
@@ -828,29 +879,38 @@ void SimulationRunner::LoadPlugins(const Entity _entity,
               << plugin.EntityType() << "]" << std::endl;
     }
 
-    // Skip plugins that do not match the provided entity
-    if (entity != _entity)
-      continue;
-
-    std::optional<SystemPluginPtr> system;
+    if (kNullEntity != entity)
+    {
+      this->LoadPlugin(entity, plugin.Filename(), plugin.Name(), plugin.Sdf());
+    }
+    else
     {
-      std::lock_guard<std::mutex> lock(this->systemLoaderMutex);
-      system = this->systemLoader->LoadPlugin(plugin.Filename(), plugin.Name(),
-                                              nullptr);
+      ignwarn << "Attempting to attach plugin [" << plugin.Name() <<
+        " to an unknown entity. Plugin won't be loaded" << std::endl;
```

Comment:
I think the reason we weren't printing a warning before is that we support specifying plugins for entities that will be spawned at runtime. So this may succeed eventually.

Reply:
I don't know that this is true.  We don't hold a queue of systems to be re-attempted later, so if the entity isn't available at this point, I don't think it will ever be tried again?

Reply:
This is used (at least used to be?) by SubT for example to add plugins to entities that are spawned while simulation is running.

Whenever a new entity is created, `events::LoadPlugins` triggers this function to see if there are any pending plugins for that entity.